src/sources/aws_kinesis_firehose/mod.rs

```diff
@@ -0,0 +1,329 @@
+use crate::{
+    config::{DataType, GlobalOptions, SinkDescription, SourceConfig},
+    shutdown::ShutdownSignal,
+    tls::{MaybeTlsSettings, TlsConfig},
+    Pipeline,
+};
+use async_trait::async_trait;
+use futures::{compat::Future01CompatExt, FutureExt, TryFutureExt};
+use serde::{Deserialize, Serialize};
+use std::net::SocketAddr;
+
+pub mod errors;
+mod filters;
+mod handlers;
+mod models;
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct AwsKinesisFirehoseConfig {
+    address: SocketAddr,
+    access_key: Option<String>,
+    tls: Option<TlsConfig>,
+}
+
+#[typetag::serde(name = "aws_kinesis_firehose")]
+#[async_trait]
+impl SourceConfig for AwsKinesisFirehoseConfig {
+    fn build(
+        &self,
+        _name: &str,
+        _globals: &GlobalOptions,
+        _shutdown: ShutdownSignal,
+        _out: Pipeline,
+    ) -> crate::Result<super::Source> {
+        unimplemented!()
+    }
+
+    async fn build_async(
+        &self,
+        _: &str,
+        _: &GlobalOptions,
+        shutdown: ShutdownSignal,
+        out: Pipeline,
+    ) -> crate::Result<super::Source> {
+        let svc = filters::firehose(self.access_key.clone(), out);
+
+        let tls = MaybeTlsSettings::from_config(&self.tls, true)?;
+        let listener = tls.bind(&self.address).await?;
+
+        let fut = async move {
+            let _ = warp::serve(svc)
+                .serve_incoming_with_graceful_shutdown(
+                    listener.accept_stream(),
+                    shutdown.clone().compat().map(|_| ()),
+                )
+                .await;
+            // We need to drop the last copy of ShutdownSignalToken only after server has shut down.
+            drop(shutdown);
+            Ok(())
+        };
+        Ok(Box::new(fut.boxed().compat()))
+    }
+
+    fn output_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn source_type(&self) -> &'static str {
+        "aws_kinesis_firehose"
+    }
+}
+
+inventory::submit! {
+    SinkDescription::new_without_default::<AwsKinesisFirehoseConfig>("aws_kinesis_firehose")
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::shutdown::ShutdownSignal;
+    use crate::{
+        config::{GlobalOptions, SourceConfig},
+        event::{Event, LogEvent},
+        log_event,
+        test_util::{collect_ready, next_addr, wait_for_tcp},
+        Pipeline,
+    };
+    use chrono::{DateTime, SubsecRound, Utc};
+    use flate2::{read::GzEncoder, Compression};
+    use futures::compat::Future01CompatExt;
```

Comment:
Not sure if they are accurate, but seems there are some lints still failing or giving warnings that would be good to fix?